aten/src/ATen/TensorIndexing.h

```diff
@@ -237,23 +238,23 @@ static inline Tensor applySelect(
     int64_t index,
     int64_t real_dim,
     const at::Device& /*self_device*/,
-    const c10::optional<IntArrayRef>& self_sizes) {
+    const c10::optional<SymIntArrayRef>& self_sizes) {
   // See NOTE [nested tensor size for indexing]
   if (self_sizes.has_value()) {
     TORCH_CHECK_INDEX(
         !(index == 0 && dim == 0 && self_sizes->size() == 0),
         "invalid index of a 0-dim tensor. ",
         "Use `tensor.item()` in Python or `tensor.item<T>()` in C++ to convert a 0-dim tensor to a number");
 
-    int64_t size = (*self_sizes)[dim];
+    auto size = (*self_sizes)[dim];
     TORCH_CHECK_INDEX(
-        index >= -size && index < size,
+        size >= -index && size > index,
         "index ",
         index,
         " is out of bounds for dimension ",
         real_dim,
         " with size ",
-        size);
+        size.guard_int(__FILE__, __LINE__));
```

Comment:
@ezyang The guard here should be fine as the error message is never evaluated right?

Reply:
Yes, is fine. But cc @Chillee was talking about making it so we always print the concrete values when printing symints, which would have made direct size print ok too.